docarray/doc_index/backends/weaviate_doc_index.py

```diff
@@ -0,0 +1,594 @@
+
+        return [self._parse_weaviate_result(doc) for doc in docs]
+
+    def _filter_batched(
+        self, filter_queries: Any, limit: int
+    ) -> Union[List[DocumentArray], List[List[Dict]]]:
+        qs = [
+            self._client.query.get(self._db_config.index_name, self.properties)
+            .with_additional("vector")
+            .with_where(filter_query)
+            .with_limit(limit)
+            .with_alias(f'query_{i}')
+            for i, filter_query in enumerate(filter_queries)
+        ]
+
+        batched_results = self._client.query.multi_get(qs).do()
+
+        return [
+            [self._parse_weaviate_result(doc) for doc in batched_result]
+            for batched_result in batched_results["data"]["Get"].values()
+        ]
+
+    def _find(
+        self,
+        query: np.ndarray,
+        search_field: str,
+        limit: int,
+        score_name: Literal["certainty", "distance"] = "certainty",
+        score_threshold: Optional[float] = None,
+    ) -> _FindResult:
+        index_name = self._db_config.index_name
+        if search_field:
+            logging.warning(
+                'Argument search_field is not supported for WeaviateDocumentIndex. Ignoring.'
+            )
+        near_vector = {
+            "vector": query,
+        }
+        if score_threshold:
+            near_vector[score_name] = score_threshold
+
+        results = (
+            self._client.query.get(index_name, self.properties)
+            .with_near_vector(
+                near_vector,
+            )
+            .with_limit(limit)
+            .with_additional([score_name, "vector"])
+            .do()
+        )
+
+        return self._format_response(results["data"]["Get"][index_name], score_name)
+
+    def _format_response(
+        self, results, score_name
+    ) -> Tuple[List[DocumentArray], List[float]]:
+        """
+        Format the response from Weaviate into a Tuple of DocumentArray and scores
+        """
+        da_class = DocumentArray.__class_getitem__(
+            cast(Type[BaseDocument], self._schema)
+        )
+
+        documents = []
+        scores = []
+
+        for result in results:
+            score = result["_additional"][score_name]
+            scores.append(score)
+
+            document = self._parse_weaviate_result(result)
+            documents.append(self._schema.from_view(document))
+
+        return da_class(documents), scores
+
+    def _find_batched(
+        self,
+        queries: Sequence[np.ndarray],
+        search_field: str,
+        limit: int,
+        score_name: Literal["certainty", "distance"] = "certainty",
+        score_threshold: Optional[float] = None,
+    ) -> _FindResultBatched:
+        if search_field != '':
+            logging.warning(
+                'Argument search_field is not supported for WeaviateDocumentIndex. Ignoring.'
+            )
+
+        qs = []
+        for i, query in enumerate(queries):
```

Comment:
Can weaviate to a native batch operation here, without looping over the queries?